src/sources/http_client/client.rs

```diff
@@ -350,11 +350,8 @@ impl http_client::HttpClientContext for HttpClientContext {
                     }
                 }
                 Event::Trace(ref mut trace) => {
-                    if let Some(source_type_key) = log_schema().source_type_key() {
-                        trace.insert(
-                            source_type_key.to_string(),
-                            Bytes::from(HttpClientConfig::NAME),
-                        );
+                    if let Some(source_type_key) = log_schema().source_type_key_target_path() {
+                        trace.insert(source_type_key, Bytes::from(HttpClientConfig::NAME));
```

Comment:
Can this use `maybe_insert`?

Reply:
Yes, certainly. It just requires a bit of extra work in `log_schema.rs`. I will follow up with another PR shortly.